### PR TITLE
Add XI CLI commands

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need a CLI command to (list and) index Moodle courses and content in the Experience Index.

## Proposal

- [x] add `xi list` commands
- [x] add `xi index` commands
- [x] write tests
- [x] add explicit error message for moodle login issues
- [x] ~xi routes should be protected behind authentication~ postponed this in a new PR

### CODE REVIEW

File: src/api/core/warren/xi/client.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ async def get(self, object_id: Union[UUID, IRI]) -> Union[ExperienceRead, None]:
             return None
 
         response.raise_for_status()
-        return ExperienceRead.construct(**response.json())
+        return ExperienceRead(**response.json())
```
Comment (quitterie-lcs) [inline src/api/core/warren/xi/client.py:112]: Why do you validate the response ?

Comment (jmaupetit) [reply src/api/core/warren/xi/client.py:112]: Pydantic's models `construct` method does not parse nested models for performance reasons. If we want a complete `ExperienceRead` instance while getting an experience from the XI, we need to parse AND validate API
response. This decreases performances but has a reasonable trade-off.

File: src/api/core/warren/cli.py [hunk 2]
```diff
@@ -176,3 +184,129 @@ def indicator_compute(ctx: click.Context, indicator: str, cache: bool):
     click.echo(
         result.json() if issubclass(compute_annotation, BaseModel) else str(result)
     )
+
+
+# -- EXPERIENCE INDEX (AKA XI) COMMAND --
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/api/core/warren/cli.py:189]: There's only one `xi` command but multiple sub-commands. The same applies for many commands. :wink:

**Review by wilbrdt — COMMENTED** [review]
Alt text
Looks good to me! Only fixing some typos

Note: don't forget to check coverage of `cli.py` (a test is probably missing for the `xi list content` command)

**Comment by jmaupetit** [comment]
> Note: don't forget to check coverage of `cli.py` (a test is probably missing for the `xi list content` command)

Well spotted! I've added a test for this.

**Review by wilbrdt — APPROVED** [review]

**Pull request merged**